satellite/console/service.go

```diff
@@ -141,6 +119,37 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		return nil, errs.New(internalErrMsg)
 	}
 
+	// TODO: remove after vanguard release
+	var registrationToken *RegistrationToken
+	if user.PartnerID != "" {
```

Comment:
May be can handle this logic in a separete function as it's getting to verbose in my opinion, and `if/else` blocks are harder to read

Reply:
We can consider creating referral service to encapsulate related logic there. It can be agreat help in future when we decide to move out tardigrade specific code from our repo

Reply:
```
struct ReferralService {
}

func (service *ReferralService) CreateUser(...params) {
    // check partnerID and RegTokenID
    regtokens.Insert(newRegToken)
    satellite.Console.Service.CreateUser(params, newRegToken)
}
```
also it would be nice to create new server on subdomain of satellite and store there new api endpoints to marketing stuf

Reply:
it would make sense to place regtokens logic to ReferralService from console

Reply:
That's a great point! We are planning to move the entire referral logic out of the satellite so we can achieve cross-satellite referrals https://github.com/storj/storj/pull/2357. Also the change seems to be bigger than the scope of this PR

Reply:
@VinozzZ definitely, we should plan some 'refactoring sprint' and do it. Kyiv team would be glad to participate in it